src/utils/barcode.ts

```diff
@@ -0,0 +1,86 @@
+import * as bwipjs from "bwip-js";
+import { sequenceS } from "fp-ts/lib/Apply";
+import { Either } from "fp-ts/lib/Either";
+import { taskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import * as JsBarcode from "jsbarcode";
+import { DOMImplementation, XMLSerializer } from "xmldom";
+
+export interface IBarcodeOutput {
+  png: string;
+  pngMimeType: string;
+  svg: string;
+  svgMimeType: string;
+}
+
+interface IBwipOptions {
+  bcid: string;
+  text: string;
+  includetext: boolean;
+}
+
+const toBase64Png = (options: IBwipOptions) =>
+  tryCatch(
+    () =>
+      new Promise<string>((resolve, reject) =>
+        bwipjs.toBuffer(options, (err, png) => {
+          if (err == null) {
+            return resolve(png.toString("base64"));
+          } else {
+            return reject(new Error(`Cannot generate png barcode|${err}`));
+          }
+        })
+      ),
+    errs => new Error(`Cannot generate png barcode|${errs}`)
+  );
+
+const toBase64Svg = (text: string) =>
```

Comment:
can you try to generate the svg using bwjp as well since it has no external / native deps ?
https://github.com/metafloor/bwip-js/blob/master/examples/svg.html#L150